src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java

```diff
@@ -48,38 +50,74 @@ public WithHoldingOrderLineUpdateInstanceStrategy(InventoryItemManager inventory
-
-      holder.createStoragePatchOrderLineRequest(StoragePatchOrderLineRequest.PatchOrderLineOperationType.REPLACE_INSTANCE_REF, newInstanceId);
-
-      return switch (replaceInstanceRef.getHoldingsOperation()) {
-        case MOVE -> moveHoldings(holder, newInstanceId, requestContext);
-        case FIND_OR_CREATE -> findOrCreateHoldingsAndUpdateItems(holder, newInstanceId, requestContext)
-          // TODO: onSuccess or compose ???
-          .onSuccess(v -> deleteAbandonedHoldings(replaceInstanceRef.getDeleteAbandonedHoldings(), holder.getStoragePoLine(), requestContext));
-        case CREATE -> createHoldingsAndUpdateItems(holder, newInstanceId, requestContext)
-          .onSuccess(v -> deleteAbandonedHoldings(replaceInstanceRef.getDeleteAbandonedHoldings(), holder.getStoragePoLine(), requestContext));
-        default -> Future.succeededFuture();
-      };
-    } else {
+    ReplaceInstanceRef replaceInstanceRef = holder.getPatchOrderLineRequest().getReplaceInstanceRef();
+    if (replaceInstanceRef == null) {
       return Future.succeededFuture();
     }
+    String newInstanceId = replaceInstanceRef.getNewInstanceId();
+
+    holder.createStoragePatchOrderLineRequest(StoragePatchOrderLineRequest.PatchOrderLineOperationType.REPLACE_INSTANCE_REF, newInstanceId);
+    PoLine poLine = holder.getStoragePoLine();
+
+    return switch (replaceInstanceRef.getHoldingsOperation()) {
+      case MOVE -> moveHoldings(poLine, newInstanceId, requestContext);
+      case FIND_OR_CREATE -> findOrCreateHoldingsAndUpdateItems(holder, newInstanceId, requestContext)
+        .compose(v -> deleteAbandonedHoldings(replaceInstanceRef.getDeleteAbandonedHoldings(), poLine, requestContext));
+      case CREATE -> createHoldingsAndUpdateItems(holder, newInstanceId, requestContext)
+        .compose(v -> deleteAbandonedHoldings(replaceInstanceRef.getDeleteAbandonedHoldings(), poLine, requestContext));
+      default -> Future.succeededFuture();
+    };
   }
 
 
-  private Future<Void> moveHoldings(OrderLineUpdateInstanceHolder holder, String newInstanceId, RequestContext requestContext) {
-    return pieceStorageService.getPiecesByPoLineId(PoLineCommonUtil.convertToCompositePoLine(holder.getStoragePoLine()), requestContext)
-      .map(pieces -> extractUniqueHoldingIds(pieces, holder.getStoragePoLine().getLocations()))
-      .compose(holdingIds -> {
-        holdingIds.forEach(holdingId -> holder.addHoldingRefsToStoragePatchOrderLineRequest(holdingId, holdingId));
-        return inventoryHoldingManager.getHoldingsByIds(holdingIds, requestContext);
-      })
-      .compose(holdings -> {
-        removeHoldingUnrecognizedFields(holdings);
-        return inventoryHoldingManager.updateInstanceForHoldingRecords(holdings, newInstanceId, requestContext);
-      });
+  private Future<Void> moveHoldings(PoLine poLine, String newInstanceId, RequestContext requestContext) {
+    CompositePoLine compPOL = PoLineCommonUtil.convertToCompositePoLine(poLine);
```

Comment:
can we use method without using its class. Sometimes line is long that is affecting to understand the code. If yes, consider other places as well

Reply:
for example
`var locationContext = RequestContextUtil.createContextWithNewTenantId(requestContext, entry.getKey());`

Reply:
Let's agree on max line length. I'm keeping < 140. Static imports are source of more confusion

Reply:
per folio code style, max line is 132, but any code shouldn't be longer (at least <120)

Reply:
That's interesting. I have horizontal scroll in almost all files. I'll keep in mind.